hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/TableCountTask.java

```diff
@@ -69,18 +75,18 @@ public TableCountTask(GlobalStatsDao globalStatsDao,
    */
   @Override
   public Pair<String, Boolean> reprocess(OMMetadataManager omMetadataManager) {
+    HashMap<String, Long> objectCountMap = initializeCountMap();
     for (String tableName : getTaskTables()) {
       Table table = omMetadataManager.getTable(tableName);
       try (TableIterator keyIter = table.iterator()) {
         long count = getCount(keyIter);
-        ReconUtils.upsertGlobalStatsTable(sqlConfiguration, globalStatsDao,
```

Comment:
You mentioned:

> With that I didn't find upsert query or anything similar, so I batched the updates and insert separately. This could be improved in the future, but for now this to cause a problem is an event with a small probability of occurrence.

Do you mean the `upsert` here? Would you elaborate on what problem could this raise when we switch to `map.put`?

Reply:
I wasn't clear with this previously, sorry. So no problem could raise if I use a map, that is why I changed to that. The goal in this patch was that the tasks should write in batches to their DB, so one batch is either fully written in the DB or not at all, so we don't have a chance that some events from this batch are processed and some aren't. With the RocksDB this could be done easily, but with this we can't upsert in batches. So I write the updates and the inserts in separate batches in the DB. I meant with that sentence that there is a really small chance that the updates are committed in the DB and something goes wrong before we commit the inserts.